src/components/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,29 @@
+import React, {useEffect, useState} from 'react';
+import {useEasyauth} from '../../hooks/useEasyauth/useEasyauth.jsx';
+import {getProfile} from '../../api/api.js';
+
+export const UserProfile = () => {
+  const auth = useEasyauth();
+  const [profile, setProfile] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      const token = auth.user?.access_token;
+      const response = await getProfile(token);
+      setProfile(response.data);
+    })();
+  }, [auth]);
+
+  if (!profile) {
```

Comment:
if ( auth.user?.access_token ) {
if (!profile) { return loading }
else { return profile component }
} else {
return null
}

Reply:
Done, in commit e8aa558a29dd3e816ae4bfdc0e3d5af04cd10506